cmux-tui/bindings/wait_for_go_module.py

```
@@ -0,0 +1,132 @@
+        raise GoModuleError("module and version must be non-empty")
+    if wait_seconds < 0:
+        raise GoModuleError("wait seconds must be non-negative")
+    if retry_seconds <= 0:
+        raise GoModuleError("retry seconds must be positive")
+
+    cancellation = cancel_event or threading.Event()
+    deadline = clock() + wait_seconds
+    last_error = "module is unavailable"
+    while True:
+        if cancellation.is_set():
+            raise GoModuleCancellation("Go module verification was cancelled")
```

Ruff (0.16.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)